src/javascript/DesignSystem/DatePicker/DatePicker.jsx

```diff
@@ -39,37 +39,83 @@ const DatePickerCmp = ({
     classes,
     onSelectDateTime,
     selectedDateTime,
+    disabledDays: incomingDisableDays,
     ...props
 }) => {
     const [selectedDays, setSelectedDays] = useState([]);
-    const [selectedHour, setSelectedHour] = useState('0:00');
+    const [selectedHour, setSelectedHour] = useState('00:00');
+    const isDateTime = variant === 'datetime';
 
     const locale = locales[lang] || {};
 
     const selectedDate = selectedDateTime ? [selectedDateTime] : selectedDays;
-    const isDateTime = variant === 'datetime';
+    let disabledHours = {};
+    const disabledDays = [];
+    const extractDateAndHours = (entry, boundIncludeOffset) => {
```

Comment:
Could be great to extract the functions that are here out side the component it self to make the component more readable.
like what we have for generateWeekdaysShort and getDateTime functions.

Reply:
all functions have been extracted and tested